### PR TITLE
fix - created shared function for generating Addon URL

### LINKED ISSUE #8563: Create a shared function which generates an addon URL
There are a number of places in our code where we generate an add-on URL using

```
`/addon/${addon.slug}/`
```

Some examples include `HeroRecommendation`,  `SearchResult`, `AddonReviewList`, `AddonSummaryCard`, and `AddonTitle`.

We should create a shared function (which should probably live in `src/amo/utils/index.js`) which accepts a `slug` and returns this `addonURL` and then make use of that function everywhere that we currently have this code.

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #8563 
Hey @bobsilverberg !

How does this look?

Let me know if this needs any changes and I would be more than happy to look at them!

Thanks!

### CODE REVIEW

**Comment by rbrishabh** [comment]
Hey @bobsilverberg !
I made the changes you mentioned.

Thanks for the amazing guidance! =)

File: src/amo/pages/Addon/index.js [hunk 1]
```diff
@@ -118,10 +119,11 @@ export class AddonBase extends React.Component {
         if (slugIsPositiveID(slug) || addon.slug !== slug) {
           // We only load add-ons by slug, but ID must be supported too because
           // it is a legacy behavior.
+          const url = getAddonURL(addon.slug);
```
Comment (bobsilverberg) [inline src/amo/pages/Addon/index.js]: We don't really need this `const`. You could just say:

```
url: `/${lang}/${clientApp}${getAddonURL(addon.slug)}`
```

on line 126, below.

File: tests/unit/amo/components/TestHeroRecommendation.js [hunk 2]
```diff
@@ -284,9 +285,10 @@ describe(__filename, () => {
       const shelfData = createShelfData({ addon: { ...fakeAddon, slug } });
 
       const root = render({ shelfData });
+      const addonURL = getAddonURL(slug);
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestHeroRecommendation.js]: As above, we don't really need this `const`. Line 291 could just be:
`addParamsToHeroURL({ urlString: getAddonURL(slug) }),`.

File: tests/unit/amo/components/TestSearchResult.js [hunk 3]
```diff
@@ -79,8 +80,9 @@ describe(__filename, () => {
     const addon = createInternalAddon({ ...fakeAddon, slug });
 
     const root = render({ addon });
+    const addonURL = getAddonURL(slug);
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestSearchResult.js]: As above, no need for this `const`.

File: tests/unit/amo/components/TestSearchResult.js [hunk 4]
```diff
@@ -165,16 +167,14 @@ describe(__filename, () => {
     const lang = 'fr';
     const history = createFakeHistory();
     const { store } = dispatchClientMetadata({ clientApp, lang });
+    const addonURL = getAddonURL(slug);
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestSearchResult.js]: No need for this `const`.

File: tests/unit/amo/pages/TestAddon.js [hunk 5]
```diff
@@ -507,6 +508,7 @@ describe(__filename, () => {
     const clientApp = CLIENT_APP_FIREFOX;
     const { store } = dispatchClientMetadata({ clientApp });
     const addon = fakeAddon;
+    const addonURL = getAddonURL(addon.slug);
```
Comment (bobsilverberg) [inline tests/unit/amo/pages/TestAddon.js]: In all the tests in this file, there's no need to define a `const` for `addonURL`. The call to `getAddonURL(addon.slug)` can just me made inline.

**Comment by rbrishabh** [comment]
Hey @bobsilverberg ! I was actually quite skeptical about whether I should be making an extra ```const``` for all those instances you mentioned.

Anyways, now I know what the best practices are!

Please review this when you get time and let me know if this still requires any changes, I'd be more than happy to look at it! =)

**Pull request merged**